src/main/java/org/opentripplanner/street/model/edge/StreetTransitStationCentroidLink.java

```diff
@@ -0,0 +1,51 @@
+package org.opentripplanner.street.model.edge;
+
+import org.opentripplanner.street.model.vertex.StreetVertex;
+import org.opentripplanner.street.model.vertex.TransitStationCentroidVertex;
+import org.opentripplanner.transit.model.basic.Accessibility;
+
+public class StreetTransitStationCentroidLink
+  extends StreetTransitEntityLink<TransitStationCentroidVertex> {
+
+  private final boolean isEntrance;
+
+  private StreetTransitStationCentroidLink(StreetVertex fromv, TransitStationCentroidVertex tov) {
+    super(fromv, tov, tov.getStation().getWheelchairAccessibility());
+    isEntrance = true;
+  }
+
+  private StreetTransitStationCentroidLink(TransitStationCentroidVertex fromv, StreetVertex tov) {
+    super(fromv, tov, fromv.getStation().getWheelchairAccessibility());
+    isEntrance = false;
+  }
+
+  public static StreetTransitStationCentroidLink createStreetTransitStationLink(
+    StreetVertex fromv,
+    TransitStationCentroidVertex tov
+  ) {
+    return connectToGraph(new StreetTransitStationCentroidLink(fromv, tov));
+  }
+
+  public static StreetTransitStationCentroidLink createStreetTransitStationLink(
+    TransitStationCentroidVertex fromv,
+    StreetVertex tov
+  ) {
+    return connectToGraph(new StreetTransitStationCentroidLink(fromv, tov));
+  }
+
+  public boolean isEntrance() {
+    return isEntrance;
+  }
+
+  public boolean isExit() {
+    return !isEntrance;
+  }
+
+  public String toString() {
+    return "StreetTransitStationCentroidLink(" + fromv + " -> " + tov + ")";
```

Comment:
I know that many edges use this style of toString but I don't really like it. I would prefer `ToStringBuilder`. Lets discuss in the dev meeting.

Reply:
I personally do not debug the AStar very often, but the notation here is a bit sharper than the `ToStringBuilder` would produce. Maybe the `toString()` can be implemented in the root of the hierarchy of Edge and Vertex. For example: `getClass().getSimpleName() + "(" + fromv + " -> " + tov + ")";`. In fact, the code above can be deleted, since the `Edge#toString()` does exactly as suggested (except using `getName()` not `getSimpleName()`). A more reusable approach might be needed. You probably need some way to override it, so a little more complex way to do it would make a reusable method - lik the `Path#buildString()` method.

**Some think like this**
```
  public String toString() {
    return buildToString(b -> {});
  }

  /**
   * Use this to construct a {@code toString()} in derived classes. The result will be:
   * {@code "%simple-class-name%(%fromv% -> %tov%" + body + ")"}. Note! There is no space
   * between {@code tov} and {@code body}.
   * 
   * @param body Use this callback to add local content.
   */
  protected String buildToString(Consumer<StringBuilder> body) {
    var buf = new StringBuilder();
    buf.append(getClass().getSimpleName())
      .append(" (")
      .append(fromv)
      .append(" -> ")
      .append(tov);

    if(body != null) {
      body.accept(buf);
    }
    return buf.append(')').toString();
  }
```

Reply:
I used your approach @t2gran and changed all edges to use the same toString format. This will change the output for some Edges that previously used the `ToStringBuilder`, but that shouldn't be a problem right?